src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryAppDeployer.java

```diff
@@ -140,7 +145,10 @@ public String deploy(AppDeploymentRequest request) {
 				.push(PushApplicationRequest.builder()
 					.name(name)
 					.application(request.getResource().getFile().toPath())
-					.domain(connectionProperties.getDomain())
+					.domain(domain(request))
+					.host(host(request))
+					.routePath(routePath(request))
+					.noRoute(toggleNoRoute(request))
```

Comment:
I think we can defer this for now. going ahead with the merge after polishing commit with the suggested changes.